examples/trainer_controller/README.md

```diff
@@ -0,0 +1,121 @@
+# Trainer Controller Framework
+
+A framework for controlling the trainer loop using user-defined rules and metrics.
+
+## High-level architecture
+
+- **Registration**: The registration mechanism de-couples the metrics and operators from the trainer framework. A user could implement a custom metric or operator and register it through the registration methods mentioned previously. This makes the framework highly extensible.
+
+- **Configuration**: The trainer controller configuration supplies the definition for triggers, rule, operations and metrics to orchestrate the enactment of a particular control policy. These details are split up and passed off to the respective modules by the trainer controller as shown in the figure.
+
+- **Events**: Events supply the state and arguments required for the metric handlers to perform metric computation at the events they are registered for. The framework callback lists out all event handlers with prefix `"on_"` and loads then as event handlers. Every metric declares one or more events from this list of valid handlers. These computed metric variables are stored in a global state of the trainer controller and independently picked up the operations which could potentially be triggered on an entirely different set of events. This decouples the control loop for metrics and operations. I.e. the metric could be computed on event A, while operation could be triggered on event B. The controller rules which use the metric variables from the trainer controller state are evaluated and based on the outcomes specified actions are performed.
+![High-Level Design Diagram: Trainer Controller Framework](../../architecture_records/imgs/001-arch.png)
+
+## Usage and customization
+We added a new trainer callback (see [here](https://huggingface.co/docs/transformers/v4.37.2/en/main_classes/callback)) which accepts a `training control definition` YAML file containing:
```

Comment:
how to pass this YAML file while training? can you list parameter name and show full example on how to use it with fms-hf-tuning for any one tuning type ? that parameter name can be moved to README.md and this section can be about describing YAML in more detail like you did. Moving comment above

Reply:
There is only flag that needs to be added to the CLI to avail the trainer controller features. Its explained in point 1 of this [section](https://github.com/foundation-model-stack/fms-hf-tuning/tree/1ca546d5050e1f6d4ba52441343fb04dc2423b89/examples/trainercontroller_configs#usage) in the readme.

Reply:
Addressed. The trainer controller configuration YAML is listed in point 2 of this [section](https://github.com/foundation-model-stack/fms-hf-tuning/tree/1ca546d5050e1f6d4ba52441343fb04dc2423b89/examples/trainercontroller_configs#usage).